presto-spi/src/main/java/com/facebook/presto/spi/block/ShortArrayBlockBuilder.java

```diff
@@ -120,7 +126,7 @@ private void growCapacity()
 
     private void updateDataSize()
     {
-        retainedSizeInBytes = INSTANCE_SIZE + sizeOf(valueIsNull) + sizeOf(values);
+        retainedSizeInBytes = INSTANCE_SIZE + sizeOf(valueIsNull) + sizeOf(values) + NULL_VALUE_BLOCK.getRetainedSizeInBytes();
```

Comment:
`NULL_VALUE_BLOCK` is const, remove `+ NULL_VALUE_BLOCK. getRetainedSizeInBytes()`